### PR TITLE
cdrdao: Fix Linux compilation

### PR DESCRIPTION
**Pull request by DoomHammer**
Gist logs: https://gist.github.com/5c40a554c35054c7178c

### CODE REVIEW

**Comment by sjackman** [comment]
https://gist.github.com/anonymous/5c40a554c35054c7178c#file-02-make-L164-L167
```
/usr/bin/g++-4.8 -DHAVE_CONFIG_H -I. -I. -I.. -I./../trackdb -I./../paranoia   -isystem/home/doomhammer/.linuxbrew/include -DDRIVER_TABLE_FILE=\"/home/doomhammer/.linuxbrew/Cellar/cdrdao/1.2.3/share/cdrdao/drivers\" -Os -w -pipe -march=core2 -c -o sg_err.o sg_err.cc
ScsiIf-linux.cc: In static member function ‘static ScsiIf::ScanData* ScsiIf::scan(int*, char*)’:
ScsiIf-linux.cc:287:37: error: no matching function for call to ‘stat::stat(const char [22], stat*)’
     if (stat(SYSFS_SCSI_DEVICES, &st) != 0) {
```

File: Library/Formula/cdrdao.rb [hunk 1]
```diff
@@ -20,7 +20,12 @@ class Cdrdao < Formula
   patch do
     url "https://sourceforge.net/p/cdrdao/patches/_discuss/thread/205354b0/141e/attachment/cdrdao-mac.patch"
     sha256 "ee1702dfd9156ebb69f5d84dcab04197e11433dd823e80923fd497812041179e"
-  end
+  end unless OS.linux?
```
Comment (sjackman) [inline Library/Formula/cdrdao.rb]: Please change to `if OS.mac?`

File: Library/Formula/cdrdao.rb [hunk 2]
```diff
@@ -20,7 +20,12 @@ class Cdrdao < Formula
...
+
+  patch do
+    url "https://raw.githubusercontent.com/PacBSD/abs/master/extra/cdrdao/cdrdao-1.2.3-stat.patch"
+    sha256 "ca89b7c56a376d5a9574c5757f0d372236a895334f81867ff5e1703806565bbc"
+  end if OS.linux?
```
Comment (sjackman) [inline Library/Formula/cdrdao.rb]: Ditto.

**Comment by sjackman** [comment]
Is disabling both patches necessary?

**Comment by DoomHammer** [comment]
Um, I only disabled one mac-specific patch. The other one I have added. It is necessary to compile under Linux :)

**Comment by sjackman** [comment]
Ah, yes. Sorry. So many pull requests. Which is a good thing!

**Comment by sjackman** [comment]
Has the Linux patch been submitted to the upstream author?

**Comment by sjackman** [comment]
For future contributions, the preferred commit message summary (the first line) is:
`cdrdao: Fix for Linuxbrew`

**Comment by sjackman** [comment]
Merged. Thanks for your contribution, Piotr!

**Comment by DoomHammer** [comment]
The patch is from a BSD distribution so I believe it has already been posted upstream.

**Pull request closed without merging**